packages/editor/src/lib/editor/Editor.ts

```diff
@@ -676,16 +676,24 @@ export class Editor extends EventEmitter<TLEventMap> {
 		)
 		this.disposables.add(this.history.dispose)
 
-		this.history.ignore(() => {
-			this.store.ensureStoreIsUsable()
-
-			// clear ephemeral state
-			this._updateCurrentPageState({
-				editingShapeId: null,
-				hoveredShapeId: null,
-				erasingShapeIds: [],
-			})
-		})
+		this.run(
+			() => {
+				this.store.ensureStoreIsUsable()
+
+				// clear ephemeral state
+				this.run(
```

Comment:
we don't need the nested `this.run` here, we're already in an ignored section

Reply:
actually, this reminds me - if you accidentally have a nested `run` - does this cause any problems? should it? (i don't think it should, but worth pondering over)